app/javascript/components/map-menu/components/sections/my-gfw/component.jsx

```diff
@@ -0,0 +1,277 @@
+import cx from 'classnames';
+import sortBy from 'lodash/sortBy';
+
+import MyGFWLogin from 'components/mygfw-login';
+import Button from 'components/ui/button/button-component';
+import Dropdown from 'components/ui/dropdown';
+import Icon from 'components/ui/icon/icon-component';
+import Pill from 'components/ui/pill';
+import editIcon from 'assets/icons/edit.svg';
+import tagIcon from 'assets/icons/tag.svg';
+import subscribedIcon from 'assets/icons/subscribed.svg';
+import logoutIcon from 'assets/icons/logout.svg';
+import screenImg1x from 'assets/images/aois/single A.png';
+import screenImg2x from 'assets/images/aois/single A @2x.png';
+
+import './styles.scss';
+
+class MapMenuMyGFW extends PureComponent {
+  constructor() {
+    super();
+    this.state = {
+      activeTags: {}
+    };
+  }
+
+  setTags() {
+    const { areas } = this.props;
+    const tags = {};
+    if (areas) {
+      areas.forEach(area =>
+        area.tags.forEach(tag => {
+          if (tags[tag] === undefined) {
+            tags[tag] = false;
+          }
+        })
+      );
+    }
+    this.setState({ activeTags: tags });
+  }
+
+  componentDidMount() {
+    this.setTags();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { location, clearActiveArea } = this.props;
+    const { location: prevLocation } = prevProps;
+
+    if (location.type !== 'aoi' && prevLocation.type === 'aoi') {
+      clearActiveArea();
+    }
+  }
+
+  renderLoginWindow() {
+    const { isDesktop } = this.props;
+    return (
+      <div className="aoi-header">
+        {isDesktop && <h3 className="title-login">Please log in</h3>}
+        <p>
+          Log in is required so you can view, manage, and delete your Areas of
+          Interest.
+        </p>
+        <p>
+          Creating an Area of Interest lets you customize and perform an
+          in-depth analysis of the area, as well as receiving email
+          notifications when new deforestation alerts are available.
+        </p>
+        <MyGFWLogin className="mygfw-login" />
+      </div>
+    );
+  }
+
+  renderNoAreas() {
+    const { isDesktop } = this.props;
+    return (
+      <div className="aoi-header">
+        {isDesktop && (
+          <h2 className="title-no-aois">
+            You haven&apos;t created any Areas of Interest yet
+          </h2>
+        )}
+        <p>
+          Creating an Area of Interest lets you customize and perform an
+          in-depth analysis of the area, as well as receiving email
+          notifications when new deforestation alerts are available.
+        </p>
+        {/* TODO: implement saveToAOIs prompts */}
+        <Button theme="theme-button-small">Learn how</Button>
+      </div>
+    );
+  }
+
+  renderAreas() {
+    const { isDesktop, areas, activeArea, goToAOI, onEditClick } = this.props;
+    const activeTags = Object.keys(this.state.activeTags).filter(
+      tag => this.state.activeTags[tag]
+    );
+    const activeAreas = sortBy(
```

Comment:
You have a lot of logic in the components. You we move these to components and add selectors for things like this?